### DISCUSSION TITLE
Multiple moulti in parallel

### DISCUSSION THREAD

**Discussion by huyz** [xavierog/moulti#9] [Q&A] [answered]
For my Ansible scripts, I actually run playbooks in parallel in separate terminal tabs (because my machines are more like pets than cattle).

But I get the following error:

```bash
Fatal: cannot listen on /Users/huyz/Library/Caches/moulti/moulti-huyz-default.socket (with backlog=100 and blocking=False): [Errno 48] Address already in use
```

Is there a simple way to run multiple moultis at the same time without conflict?

**Comment by xavierog** [comment, marked as answer]
Each Moulti instance has a name, and that name defaults to `default`.
To run multiple Moulti instances concurrently, they need to have different names.
Typically: `MOULTI_INSTANCE=ansible-$$ moulti run -- ansible-playbook ...`

**Reply by xavierog** [reply]
Update: as of Moulti 1.30.0, this is no longer required.
Excerpt from the [changelog for v1.30.0](https://github.com/xavierog/moulti/releases/tag/v1.30.0):
> - `moulti run` now suffixes the instance name with its process id (e.g. `default-1234` instead of `default`).
>  This helps prevent clashes and makes it possible to run various tools concurrently without having to worry about such details.
>  This behaviour can be prevented by setting the `MOULTI_RUN_NO_SUFFIX` environment variable or using the `-n` / `--no-suffix` command-line option.
>  The behaviour of `moulti init` remains unchanged.
> - `moulti run`, `moulti diff run`, `moulti manpage run`: it is no longer mandatory to use `--` to separate Moulti arguments and the > command to be run.

Otherly put: `moulti run ansible-playbook ...` should work out of the box: no need to specify `MOULTI_INSTANCE`, no need to use `--`.
